src/features/PokemonsList/pokemonSlice.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  createAsyncThunk,
+  createSlice,
+  type PayloadAction,
+} from '@reduxjs/toolkit';
+import type { RootState } from '../../app/providers/store.ts';
+
+interface PokemonState {
+  pokemons: { name: string }[];
+}
+
+const initialState: PokemonState = {
+  pokemons: [],
+};
+
+export const exportToCSV = createAsyncThunk<
+  unknown,
+  undefined,
+  { rejectValue: string; state: RootState }
+>('pokemons/exportToCSV', async (_, { getState, rejectWithValue }) => {
+  const state = getState();
+  const pokemons = state.pokemons.pokemons;
+
+  if (pokemons.length === 0) {
+    return rejectWithValue('Нет покемонов для экспорта');
+  }
+
+  try {
+    const csvContent = pokemons
+      .map((p, index) => `${index + 1}. Name: ${p.name}`)
+      .join('\n');
+    const blob = new Blob([csvContent], { type: 'text/csv;charset=utf-8;' });
+    const url = URL.createObjectURL(blob);
+    const link = document.createElement('a');
+    link.href = url;
+    link.setAttribute('download', `${pokemons.length}_items.csv`);
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+    URL.revokeObjectURL(url);
+  } catch (error) {
```

Comment:
The exportToCSV thunk directly manipulates the DOM (document.createElement, appendChild, click) and uses browser-specific APIs. UI-related logic should be separated from Redux state management to keep concerns isolated and avoid issues with testing or SSR. Consider moving the file download logic to a utility function or component.